Add saving other metadata from songs.

X-Ryl669 opened the pull request:
Allow browsing by genre.
Fix the errors in PR #487 
@alex-phillips This is based on the current master branch, and I've removed anything not related to the genre + metadata stuff.

phanan commented:
Sorry guys, but there are a couple of issues with this PR 

* Bug(s) – please see the reviews. 
* Leftovers (`kutr` font and icons). 
* There's no proper test apparently. No validation of data creation/modification can be found. 
* AWS integration, which is mentioned in the original PR.

phanan commented:
Actually, can you hold this off a bit? I'm doing a massive code linting on the JavaScript part (using [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue)) and don't want you guys to have to fix all the conflicts again.

X-Ryl669 commented:
@phanan Let us know when you think things have settled enough for us to resume working on this.

phanan commented:
Yes please.

alex-phillips commented:
@X-Ryl669 Can you go ahead and clean this up? This PR is getting kind of messy with 2 implementing the same features. I would clean this up, but have no control over your branch.

Ideally it would be great if you could PR my fork. I'll see if I can pick and choose your changes that seem relevant to this feature and implement them into mine.

X-Ryl669 commented:
@alex-phillips: I don't know how to do this correctly. I've started from your fork, but there was too many commit and I could not submit a PR to phanan's branch. So, what I did, is restarted from my own's master branch (with many changes already), and applied the diff from your fork. Then I made many modifications (to get it to build and work, split the genre into a real SQL table, add GUI, API changes, etc...).

Once I was confident with my code, I've cherry-picked all the commit to a new branch based on phanan's master branch, and submitted/fixed this PR until Travis accepted it. 

Back to your remark, since you're rebasing from phanan's branch in your fork, and since this PR includes 100% of your work (but based on phanan's master), I don't know what's the best way to work together. Either I have to bootstrap again from your fork and then submit a PR to you, than you'll have to submit a PR to phanan... It's not easy and very time consuming, either I give you write access to my fork's branch (is it possible ?) and you fix the conflict based on what I did on my master branch (but then it's time consuming for you), I don't know what's best.

I've already fixed all conflicts in my fork, but since I'm so far from phanan's branch, it's hard to backport them. Right now, I'm working on an Android's plugin for Kutr/Koel on Orpheus' music player, and I prefer to spend time on this instead of redoing the PR to keep up with the changes.

So, tell me what you prefer.

alex-phillips commented:
@X-Ryl669 Let me rebase and fix my PR and then see if I can manually add the changes you've made. That way we're starting from @phanan base master with my changes for this feature (with no additional alterations not intended to be in master) and then gradually see if I can add your additions by looking at the changes in this PR. I suggest we close this PR and stick to the other since, as you said, your master has many additional changes that are hard to back pedal. I can then give you access to my fork in necessary for you to manually add anything I may have missed.

X-Ryl669 commented:
Ok, please go for it. Beware that, since I've changed the DB format (new Genre table), you need to remove the migration files you've done and use mine that includes both yours and mine. Except for this, it's almost 1:1 from your code (except for genre in song table, obviously).
